src/Psl/Iter/last_key_opt.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Iter;
+
+use Psl\Option\Option;
+
+/**
+ * Returns the first key of an iterable wrapped in {@see Option::some},
+ * if the iterable is empty, {@see Option::none} will be returned.
+ *
+ * @template Tk
+ * @template Tv
+ *
+ * @param iterable<Tk, Tv> $iterable
+ *
+ * @return Option<Tk>
+ */
+function last_key_opt(iterable $iterable)
+{
+    $last = Option::none();
+    foreach ($iterable as $k => $_) {
+        $last = $k;
+    }
+
+    return $last instanceof Option ? $last : Option::some($last);
```

Comment:
this breaks static analysis.

if the given type is `iterable<string, Option<bool>>` psalm assume the return type is `Option<Option<bool>>`, you are breaking that here by making it `Option<bool>`. meaning, regardless of the type of `$last` it should be returned as is.

```suggestion
    $last = Option::none();
    foreach ($iterable as $k => $_) {
        $last = Option::some($k);
    }

    return $last;
```

Reply:
true. i did not want to initialize Option for each value but you're right